test/model/src/include/model/driver/kv_workload_runner.h

```diff
@@ -66,32 +67,32 @@ class kv_workload_runner {
      * kv_workload_runner::run --
      *     Run the workload in the model.
      */
-    inline void
+    inline std::vector<int>
     run(const kv_workload &workload)
     {
+        std::vector<int> ret;
         for (size_t i = 0; i < workload.size(); i++)
-            run_operation(workload[i]);
+            ret.push_back(run_operation(workload[i]));
+        return ret;
     }
 
     /*
      * kv_workload_runner::run_operation --
      *     Run the given operation.
      */
-    inline void
+    inline int
     run_operation(const operation::any &op)
     {
-        std::visit(
-          [this](auto &&x) {
-              int ret = do_operation(x);
-              /*
-               * In the future, we would like to be able to test operations that can fail, at which
-               * point we would record and compare return codes. But we're not there yet, so just
-               * fail on error.
-               */
-              if (ret != 0 && ret != WT_NOTFOUND)
-                  throw wiredtiger_exception(ret);
-          },
-          op);
+        int ret;
+        std::visit([this, &ret](auto &&x) { ret = do_operation(x); }, op);
+
+        /*
+         * Allow only certain return codes to propagate through. Turn unexpected error codes into
+         * exceptions.
+         */
+        if (ret != 0 && ret != WT_NOTFOUND)
+            throw wiredtiger_exception(ret);
```

Comment:
Why convert to exceptions? I thought that was temporary until return codes can be verified. Won't the return code comparison reject it and give a better explanation for the failure?

Reply:
I was afraid that if an operation fails in some other way, it may have downstream impact, which we are not currently handling. But I just tested the case that I was worried about the most in `WT_ROLLBACK`, and it's behaving okay, so I removed this check in [31122f3](https://github.com/wiredtiger/wiredtiger/pull/10329/commits/31122f357c1ab74e8d8c8831a92eabf6922173ce).